### PR TITLE
Add a draft of the changes coming to config processing in Oct 2019

### PR DESCRIPTION
**Pull request by ndintenfass**

### CODE REVIEW

**Review by gordonsyme — APPROVED** [review]
I tweaked some of the error messages and config samples.

lgtm

File: config-processing-changelog/2019-10-10.md [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+* Fixed a bug where a `null` default value for a parameter were accepted as
+  valid defaults. Parameters declared with a `null` default value will always
+  need to be passed in invocations, the following will no longer be accepted:
+  ```yaml
+  jobs:
+    my-job:
+      parameters:
+        foo:
+          type: string
+          default:
```
Comment (eddiewebb) [inline config-processing-changelog/2019-10-10.md:39]: is `default: ""` still valid? and will that evaluate as falsey?

File: config-processing-changelog/2019-10-10.md [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+    my-job:
+      machine: true
+      steps:
+        - run: echo hello
+
+  workflows:
+    my-workflow:
+      jobs:
+        - my-job:
+            unknown_arg: "hi" # this will error, `unknown_arg` is not a formal parameter for my-job
```
Comment (eddiewebb) [inline config-processing-changelog/2019-10-10.md:69]: weird, I feel like this already threw an error.

**Review by eddiewebb — DISMISSED** [review]

File: config-processing-changelog/2019-10-10.md (hunk 1 above)
Comment (stig) [reply config-processing-changelog/2019-10-10.md:39]:
```clojure
user> (some? "")
true
```

so `default: ""` should work just fine.

File: config-processing-changelog/2019-10-10.md (hunk 2 above)
Comment (gordonsyme) [reply config-processing-changelog/2019-10-10.md:69]: There are many cases where it does not.

File: config-processing-changelog/2019-10-10.md (hunk 1 above)
Comment (gordonsyme) [reply config-processing-changelog/2019-10-10.md]: Only `null` values for `default` are affected, as documented.

Comment (gordonsyme) [reply config-processing-changelog/2019-10-10.md:39]: Only `null` values for `default` are affected, as documented.

**Comment by gordonsyme** [comment]
@eddiewebb why did you request changes in order to ask questions? That's a blocking action.

**Pull request merged**